frontend/components/wrapper/pages/CourseDetail.tsx

```diff
@@ -0,0 +1,207 @@
+'use client'
+import React, { useEffect, useState } from 'react'
+import { Session } from 'next-auth'
+
+import {
+  upsertViewingStatus,
+  fetchButtonStatus,
+  fetchCheckMarkStatuses,
+} from '../../../app/api/course/[courseId]/viewingStatus'
+import {
+  upsertFavoriteVideo,
+  fetchFavButtonStatus,
+} from '../../../app/api/course/[courseId]/favoriteVideo'
+import { CourseWithSectionsType } from '../../../types/CourseType'
+import { QuestionType } from '../../../types/QuestionType'
+import { useCustomToast } from '../../../hooks/useCustomToast'
+import { AnswerType } from '../../../types/AnswerType'
+import { CourseDetail } from '../../pages/CourseDetail'
+import Error from '../../../app/error'
+
+export function CourseDetailWrapper({
+  courseId,
+  initialCourseData,
+  session,
+  userId,
+  videoId,
+  questions,
+  answers,
+  questionId,
+}: {
+  courseId: string
+  session: Session | null
+  initialCourseData: CourseWithSectionsType
+  userId: string
+  videoId: string
+  questions?: QuestionType[]
+  answers: AnswerType[]
+  questionId?: string
+}) {
+  const { showErrorToast } = useCustomToast()
+
+  try {
+    const [courseData, setCourseData] = useState(initialCourseData)
+    const [completePercentage, setCompletePercentage] = useState(0)
+    const [watchedStatus, setWatchedStatus] = useState<Record<string, boolean>>(
+      {},
+    )
+    const [checkedStatus, setCheckedStatus] = useState<Record<string, boolean>>(
+      {},
+    )
+    const [favoritedStatus, setFavoritedStatus] = useState<
+      Record<string, boolean>
+    >({})
+    const [loadingStates, setLoadingStates] = useState<{
+      watching: boolean
```

Comment:
watchingって何のために用意してます？
確認したいです！！

Reply:
もし必要なければ
```suggestion
    const [loadingStates, setLoadingStates] = useState<boolean>()
```
みたいにできるので

Reply:
watchingは「視聴完了ボタン用のデータ（boolean）取得中のローダー表示」に使用しています！
具体的には、ページレンダリング時と、ボタン押下時に、ボタン内に表示されるローダーです。

isWatchedのほうが適切かなと思ったので、そちらに名前を変えました。

Reply:
ってなると
それぞれ別個に　useState用意したいです。

```
    const [isWatchingLoading, setIsWatchingLoading] = useState<boolean>()
    const [isFavoriteLoading, setIsFavoriteLoading] = useState<boolean>()
```
ですかね！

Reply:
ありがととうございます、修正しました！
また、その他のファイルの関連箇所も修正しました。